test/unit/decode_fixtures.test.cpp

```diff
@@ -186,9 +186,169 @@ TEST_CASE("Read feature multipolygon")
+    auto pt = f.geometry.get<mapbox::geometry::point<std::int64_t>>();
+    CHECK(pt.x == 25);
+    CHECK(pt.y == 17);
+    // it would be better to assert on a property that is unique, rather than a geometry that is unique
+}
+
+TEST_CASE("Read: missing layer version property, defaults to a specific version and properly parses")
+{
+    std::string buffer = open_tile("test/mvt-fixtures/fixtures/024/tile.mvt");
+    auto fm = mapbox::vector_tile::decode_tile<std::int64_t>(buffer);
+    CHECK(!fm.empty());
+    // not exactly sure how to test that this one works other than making sure decode_tile actually works
+}
+
+TEST_CASE("Read: layer has no features, results in an empty feature map")
+{
+    std::string buffer = open_tile("test/mvt-fixtures/fixtures/025/tile.mvt");
+    auto fm = mapbox::vector_tile::decode_tile<std::int64_t>(buffer);
+    CHECK(fm.empty());
+}
+
+TEST_CASE("Read: has an extra Value type called 'my_value' which is not handled in the property map")
+{
+    std::string buffer = open_tile("test/mvt-fixtures/fixtures/026/tile.mvt");
+    auto fm = mapbox::vector_tile::decode_tile<std::int64_t>(buffer);
+    REQUIRE(fm.size() == 1);
+    auto fc = fm["hello"];
+    REQUIRE(fc.empty()); // my_value does not make it through
+}
+
+TEST_CASE("Read: all valid property value types")
+{
+    std::string buffer = open_tile("test/mvt-fixtures/fixtures/038/tile.mvt");
+    auto fm = mapbox::vector_tile::decode_tile<std::int64_t>(buffer);
+    REQUIRE(fm.size() == 1);
+    REQUIRE(fm.end() != fm.find("hello"));
+    auto fc = fm["hello"];
+    REQUIRE(fc.size() == 1);
+    auto f = fc[0];
+    REQUIRE(f.properties.size() == 7);
+
+    REQUIRE(f.properties.end() != f.properties.find("string_value"));
+    auto string_value = f.properties["string_value"];
+    CHECK(string_value.is<std::string>());
+    CHECK(string_value.get<std::string>() == "ello");
+
+    REQUIRE(f.properties.end() != f.properties.find("bool_value"));
+    auto bool_value = f.properties["bool_value"];
+    CHECK(bool_value.is<bool>());
+    CHECK(bool_value.get<bool>() == true);
+
+    REQUIRE(f.properties.end() != f.properties.find("int_value"));
+    auto int_value = f.properties["int_value"];
+    CHECK(int_value.is<std::int64_t>());
+    CHECK(int_value.get<std::int64_t>() == 6);
+
+    REQUIRE(f.properties.end() != f.properties.find("double_value"));
+    auto double_value = f.properties["double_value"];
+    CHECK(double_value.is<double>());
+    CHECK(double_value.get<double>() == 1.23);
+
+    REQUIRE(f.properties.end() != f.properties.find("float_value"));
+    auto float_value = f.properties["float_value"];
+    CHECK(float_value.is<double>());
+    CHECK(float_value.get<double>() == Approx(3.1));
+
+    REQUIRE(f.properties.end() != f.properties.find("sint_value"));
+    auto sint_value = f.properties["sint_value"];
+    CHECK(sint_value.is<std::int64_t>());
+    CHECK(sint_value.get<std::int64_t>() == -87948);
```

Comment:
I noticed sint_type isn't part of the [property map](https://github.com/mapbox/vector-tile/blob/2.x/include/mapbox/vector_tile/property_map.hpp) - is this right?

Reply:
Yes, it is correct. 
https://github.com/mapbox/vtzero/blob/master/include/vtzero/property_value.hpp#L329-L349

Protozero extracts it as int64_t https://github.com/mapbox/vtzero/blob/master/include/vtzero/property_value.hpp#L78-L80